text/0626-invoke-helper.md

```diff
@@ -0,0 +1,564 @@
+
+    invokeHelper(this, DocTitle, () => {
+      return {
+        positional: [`Search Result for "${this.query}"`],
+      };
+    });
+  }
+}
+```
+
+Note how the concerns are completely separated in this version of the component.
+The polling logic is self contained, and separated from the data fetching logic.
+Both sets of logic are able to contain their lifecycles, updating based on
+changes to tracked state, and tearing down when the program destroys them. In
+the future, convenience APIs can be added to make invoking them easier to read:
+
+```js
+export default class Search extends Component {
+  @use data = fetchData(() => `www.example.com/search?query=${this.query}`);
+
+  constructor() {
+    super(...arguments);
+
+    use(this, docTitle(() => `Search Result for "${this.query}"`));
+  }
+}
+```
+
+## Detailed design
+
+This RFC proposes adding the `invokeHelper` function, imported from
+`@ember/helper`. The function will have the following interface (using
+TypeScript types for brevity and clarity):
+
+```ts
+interface TemplateArgs {
+  positional?: unknown[],
+  named?: Record<string, unknown>
+}
+
+type HelperDefinition = object;
+
+interface Helper {
+  value: unknown;
+}
+
+function invokeHelper(
+  parentDestroyable: object,
+  definition: HelperDefinition,
+  computeArgs?: (context: object) => TemplateArgs
+): Helper;
+```
+
+Let's step through the arguments to the function one by one:
+
+#### `parentDestroyable`
+
+This is the parent for the helper definition. The helper will be associated as a
+destroyable to this parent context, using the [destroyables API](https://github.com/emberjs/rfcs/blob/master/text/0580-destroyables.md),
+so that its lifecycle is tied to the parent. The only requirement of the parent
+is that it is an object of some kind that can be destroyed. If the parent has an
+owner, this owner will also be passed to the helper manager that it is invoked on.
+
+This allows helper's lifecycles to be entangled correctly with the parent, and
+encourages users to ensure they've properly handled the lifecycle of their
+helper.
+
+#### `definition`
+
+This is the helper definition. It can be any object, with the only requirement
+being that a helper manager has been associated with it via the
+[`setHelperManager` API](https://github.com/emberjs/rfcs/blob/master/text/0625-helper-managers.md#detailed-design).
+
+#### `computeArgs`
+
+This is an optional function that produces the arguments to the helper. The
+function receives the parent context as an argument, and must return an object
+with a `positional` property that is an array and/or a `named` property that is
+an object.
+
+This getter function will be _autotracked_ when it is run, so the process of
+retrieving the arguments is autotracked. If any of the values used to create the
+arguments object change, the helper will be updated, just like in templates.
+
+### Return Value
+
+The function returns a reference to the helper. The public API of the reference
```

Comment:
I don't think this is correct. You just don't get a reference to the helper, you get some wrapping value that has a `.value` property but **not** "the helper".